src/ImageSharp/Formats/Png/Zlib/DeflaterEngine.cs

```diff
@@ -464,6 +462,7 @@ private void SlideWindow()
         /// </summary>
         /// <param name="curMatch">The current match.</param>
         /// <returns>True if a match greater than the minimum length is found</returns>
+        [MethodImpl(InliningOptions.HotPath)]
         private bool FindLongestMatch(int curMatch)
```

Comment:
This method currently take the longest when encoding. I need help if we are to make it any faster.

![image](https://user-images.githubusercontent.com/385879/78600027-1432ad80-784a-11ea-8476-2eb863158250.png)

Reply:
I'm up for any and all optimization approaches you can think of, feel free to experiment with the branch.

Reply:
@Sergio0694 If you have the time, do you mind experimenting with that idea?